entrypoint.d/45-geoip

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+if [[ -e /etc/GeoIP.conf ]] ; then
+    set -e
+    geoipupdate
```

Comment:
What if this fails to connect?

Would this slow down odoo boot process? :thinking:

Reply:
I need to test what happens in case of a connection failure. I assumed it will not let you start Odoo then. If you have no other mechanism in place this was my approach to make sure geoip is updated. For me the whole point of them changing their license from CC-BY-SA 4.0 to their custom EULA is that they can legally make sure you are updating the database frequently.
My intention was to make it as easy as it can be to comply with the EULA.

And yes running code in entrypoint makes the boot process slower (but i guess geoipupdate is faster than any odoo stuff ;-) )

Reply:
For example, demos are supplied with network isolation out of the box as a doodba feature. Indeed we need to cover this possibility.

First of all, just check:

- what happens if there's no network? like in demos
- how long it takes when there is network?

Depending on the results, we'll know if there's a real problem.

Reply:
I still need to test that. What do you mean by "like in demos" (i cannot find the reference demo in https://github.com/Tecnativa/doodba/blob/master/README.md nor in https://github.com/Tecnativa/doodba-scaffolding)?

Reply:
I mean https://github.com/Tecnativa/doodba#testing, where it says "it is isolated".

Reply:
Timing tests:
* ran with `time geoipupdate`
* at our office speedtest.net: 6ms ping, 300Mbps down, 50Mbps up
* ping updates.maxmind.com average 11.739ms

Result:
* with network connectivity 0,542s
* with no network connectivity (docker compose network internal true) 20,005s

Error with timeout, so it seems there is a 20 second timeout:
`error retrieving updates: error retrieving filename: error performing HTTP request: Get https://updates.maxmind.com/app/update_getfilename?product_id=GeoLite2-Country: dial tcp: lookup updates.maxmind.com on 127.0.0.11:53: read udp 127.0.0.1:37138->127.0.0.11:53: i/o timeout`

Reply:
Yikes 20s is a lot... :confused: 

Try to spawn the process, it should not affect boot time:

```suggestion
    geoipupdate &
```

Please test and report :blush:

Reply:
i tested the `&` (running the geoipupdate in background).


test commands:
with disabled maxmind
```
```

Reply:
Can you confirm also that quickly stopping the odoo container reaps the geoipupdate process properly instead of leaving it as zombie?

Reply:
yes it seems it is killed as soon as the main process (odoo) stops

tested with `docker-compose up` in one window and `while true; do date; ps fax | grep -E '(geoipupdate|odoo)' | grep -v grep; sleep 1; done` in the other pressing CTRL+C in the docker window as soon as odoo process is started.

it takes about 5 seconds for odoo to stop and geoiupdate is gone as soon as odoo is gone in ps fax

Reply:
Perfect! Thanks!